src/shared/clients/ecsClient.ts

```diff
@@ -5,70 +5,40 @@
 
 import { ECS } from 'aws-sdk'
 import { ext } from '../extensionGlobals'
+import { getLogger } from '../logger'
 import { ClassToInterfaceType } from '../utilities/tsUtils'
 
+export interface ListServicesResponse {
+    readonly services: ECS.Service[]
+    readonly continuationToken?: string
+}
+
 export type EcsClient = ClassToInterfaceType<DefaultEcsClient>
 export class DefaultEcsClient {
     public constructor(public readonly regionCode: string) {}
 
-    public async *listClusters(): AsyncIterableIterator<string> {
-        const sdkClient = await this.createSdkClient()
-        const request: ECS.ListClustersRequest = {}
-        do {
-            const response = await this.invokeListClusters(request, sdkClient)
-            if (response.clusterArns) {
-                yield* response.clusterArns
-            }
-            request.nextToken = response.nextToken
-        } while (request.nextToken)
-    }
-
-    public async *listServices(cluster: string): AsyncIterableIterator<string> {
+    public async listClusters(): Promise<ECS.Cluster[]> {
         const sdkClient = await this.createSdkClient()
-        const request: ECS.ListServicesRequest = {
-            cluster,
-        }
-        do {
-            const response = await this.invokeListServices(request, sdkClient)
-            if (response.serviceArns) {
-                yield* response.serviceArns
-            }
-            request.nextToken = response.nextToken
-        } while (request.nextToken)
+        const clusterArnList = await sdkClient.listClusters().promise()
+        const clusterReponse =  await sdkClient.describeClusters({ clusters: clusterArnList.clusterArns}).promise()
+        return clusterReponse.clusters ?? []
     }
 
-    public async *listTaskDefinitionFamilies(): AsyncIterableIterator<string> {
+    public async listServices(cluster: string): Promise<ListServicesResponse> {
         const sdkClient = await this.createSdkClient()
-        // do we also want to cover inactive? If so, would we want to use a separate function?
-        const request: ECS.ListTaskDefinitionFamiliesRequest = {}
-        do {
-            const response = await this.invokeListTaskDefinitionFamilies(request, sdkClient)
-            if (response.families) {
-                yield* response.families
+        const serviceArnList = await sdkClient.listServices({cluster: cluster}).promise()
+        try {
```

Comment:
This pagination logic is repeated throughout our code. Having a generic function that takes an AWS request and automatically 'de-paginates' the call would be great. There are some cases (albeit uncommon) that this behavior isn't wanted. So possibly we can have the wrappers expose a call with a different name (`listServicesAll`?) that automatically gets all the data. 

This is just me thinking so don't worry about trying to add any of this to your PR.

Reply:
This is a great call-out. That would make it easier to implement new services, add consistency, and require less testing. +1